clients/lsp-rust.el

```diff
@@ -860,6 +866,45 @@ them with `crate` or the crate name they refer to."
        (lambda (_) (concat "*" label "*")))
       (setq lsp-rust-analyzer--last-runnable runnable))))
 
+(defun lsp-rust-analyzer-debug (runnable)
+  "Select and run a runnable action."
+  (interactive (list (lsp-rust-analyzer--select-runnable)))
+  (unless (featurep 'dap-cpptools)
+    (user-error "You must require `dap-cpptools'."))
+  (-let (((&rust-analyzer:Runnable
+           :args (&rust-analyzer:RunnableArgs :cargo-args :workspace-root? :executable-args)
+           :label) runnable))
+    (cl-case (aref cargo-args 0)
+      ("run" (aset cargo-args 0 "build"))
+      ("test" (when (-contains? (append cargo-args ()) "--no-run")
+                (cl-callf append cargo-args (list "--no-run")))))
+    (->> (append (list (executable-find "cargo"))
+                 cargo-args
+                 (list "--message-format=json"))
+         (s-join " ")
+         (shell-command-to-string)
+         (s-lines)
+         (-keep (lambda (s)
+                  (condition-case _
```

Comment:
Nit: this could be `(ignore-error error ...)` or at least `condition-case nil`, since _ still captures the error, while `nil` doesn't.

Reply:
sure.